### PR TITLE
DevOps: Use self-hosted Linux build agents.

### PR DESCRIPTION
**Pull request by jhoerr**
* Use new CSSD-Linux-ELA agent pool
* Enable parallel job execution of packaging and deploys
* Tweak Docker files to work in ELA Linux environment.

### CODE REVIEW

File: Dockerfile.API [hunk 1]
```diff
@@ -4,19 +4,21 @@ FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS installer-env
 # https://www.reddit.com/r/docker/comments/i34thd/dotnet_core_slow_restore/g0azda1?utm_source=share&utm_medium=web2x&context=3
 # It does help, we went from 3m20s to build after changes to 1m20s
 
-COPY src/API/*.csproj ./src/API/
-COPY src/Database/*.csproj ./src/Database/
-COPY src/Models/*.csproj ./src/Models/
+COPY src/API/*.csproj ./app/src/API/
+COPY src/Database/*.csproj ./app/src/Database/
+COPY src/Models/*.csproj ./app/src/Models/
+WORKDIR  /app
```
Comment (jasonfrancis) [inline Dockerfile.API:10]: Just for posterity the changes to the `COPY` and addition of `WORKDIR` came from a handy [GitHub issue](https://github.com/dotnet/dotnet-docker/issues/644#issuecomment-423051438).

File: pipelines/azure-pipelines-template.yml [hunk 2]
```diff
@@ -16,36 +16,44 @@ resources:
...
+        version: 3.1.x
+
+    - task: DotNetCoreCLI@2
+      inputs:
+        command: test
+        projects: tests/API/Integration/Integration.csproj  
 
 - stage: Package
   displayName: Package  
   dependsOn: Test
```
Comment (jasonfrancis) [inline pipelines/azure-pipelines-template.yml:46]: Packaging is fast, but if we want to increase performance we could just do it as soon as any agent is available.

**Review by jasonfrancis — APPROVED** [review]
Looks good. It's an exciting use of our new build agent pool.
I left a comment, but it's just food for thought.

File: pipelines/azure-pipelines-template.yml (hunk 2 above)
Comment (jhoerr) [reply pipelines/azure-pipelines-template.yml]: I love this idea. Done.

**Pull request merged**